packages/regenerator-transform/src/visit.js

```diff
@@ -16,6 +16,53 @@ import * as util from "./util";
 
 let getMarkInfo = require("private").makeAccessor();
 
+// we want to convert this shorthand object method into a normal object property
+// which is a function expression. for example, this:
+//  var foo = {
+//    *bar(baz) { return 5; }
+//  }
+// should be replaced with:
+//  var foo = {
+//    bar: function*(baz) { return 5; }
+//  }
+// to do this, we will clone the parameter array and the body of the function
+// into a new FunctionExpression.
+function replaceShorthandObjectMethod(path) {
+  if (!t.isObjectMethod(path.node)) {
+    return path;
+  }
+
+  // if it's not a generator method, we don't need to do anything.
+  if (!path.node.generator) {
+    return path;
+  }
+
+  const parameters = path.node.params.map(function (param) {
+    return t.cloneDeep(param);
```

Comment:
> Are we really worried about preserving the parts of the original AST that are being replaced?

I think we are very concerned about preserving the parts of the original function parameters and function body. It seems to me that any change to the function params or body could accidentally change the semantics of the code significantly.

> How necessary are these deep clones (this one and the others)? 

I chose to deepClone because I wasn't sure how babel responds if you use a single node object in two different ASTs (the original AST and the replacement AST). It felt like it might be OK, but I couldn't find any documentation that guaranteed that it would work, so cloning felt safer.

If you would like, I can try just reusing the parameter, body, and object key nodes in the new tree rather than cloning and see if the tests pass. Let me know.